tests/openedx_tagging/core/tagging/test_system_defined.py

```diff
@@ -0,0 +1,236 @@
+""" Test the System-defined taxonomies and object tags """
+
+import ddt
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.test.testcases import TestCase, override_settings
+
+from openedx_tagging.core.tagging.system_defined_taxonomies.object_tags import (
+    ModelObjectTag,
+    UserObjectTag,
+    LanguageObjectTag,
+)
+from openedx_tagging.core.tagging.models import Tag, Taxonomy
+
+from .test_models import TestTagTaxonomyMixin
+
+test_languages = [
+    ('en', 'English'),
+    ('az', 'Azerbaijani'),
+    ('id', 'Indonesian'),
+    ('qu', 'Quechua'),
+    ('zu', 'Zulu'),
+]
+
+
+class EmptyTestClass:
+    """
+    Empty class used for testing
+    """
+
+
+class EmptyTestModel(models.Model):
+    """
+    Model used for testing
+    """
+    mi_id = models.AutoField(primary_key=True)
+
+    class Meta:
+        managed = False
+        app_label = 'oel_tagging'
+
+
+class EmptyModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+
+class NotDjangoModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestClass
+
+
+class NotIdModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestModel
+
+
+class TestUserObjectTag(UserObjectTag):
+    """
+    User ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+
+@ddt.ddt
+class TestSystemDefinedObjectTags(TestTagTaxonomyMixin, TestCase):
+    """
+    Test for generic system defined object tags
+    """
+    def test_system_defined_is_valid(self):
+        # Valid
+        assert TestUserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
+
+        # Null taxonomy
+        assert not UserObjectTag()._check_system_taxonomy()
+
+        # Not system defined taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.taxonomy)
+
+        # Not connected with the taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
```

Comment:
Can use ddt, avoid tying your tests to internal methods, and deal with the [changes I suggested above](https://github.com/open-craft/openedx-learning/pull/3/files#diff-2ecc1274a15be58e16fa617e6e5860e1b2815b6cbeffa200daea9676c07c4bfcR88) by testing this way instead:

```suggestion
@ddt.data(
    "None",
    "taxonomy",
    "user_system_taxonomy",
)
def test_system_defined_is_valid(self, taxonomy_attr):
        taxonomy = getattr(self, taxonomy_attr, None)
        object_tag =  TestUserObjectTag(taxonomy=taxonomy)
        assert not object_tag.is_valid(check_taxonomy=True, check_tag=False, check_object=False)
```

Reply:
All test has been updated to use ddt